tests/python/pants_test/option/test_options_integration.py

```diff
@@ -275,3 +277,19 @@ def test_pants_ignore_option(self):
       self.assertIn("pants_ignore = ['.*/', '/dist/', 'some/random/dir'] (from CONFIG in {})"
                     .format(config_path),
                     pants_run.stdout_data)
+
+  def test_pants_symlink_workdirs(self):
+    with temporary_dir('/tmp') as physical_workdir:
+      symlink_workdir = f'{get_buildroot()}/.pants.d'
+      physical_workdir_base = f'{physical_workdir}/workdirs'
+
+      pants_run = self.run_pants_with_workdir(
+        [f'--pants-physical-workdir-base={physical_workdir_base}',
+          'clean-all',
+          '--async']
```

Comment:
Passing async will make this racy.

Rather than checking the symlink after clean-all, you should check it after running something like `help`, and then you should check that the symlink and the destination are _gone_ after `clean-all`.

Reply:
Added directory check